toolkit/featureforms/src/main/java/com/arcgismaps/toolkit/featureforms/FeatureForm.kt

```diff
@@ -106,6 +118,119 @@ public sealed class FeatureFormEditingEvent {
         FeatureFormEditingEvent()
 }
 
+/**
+ * Indicates the navigation route within a [FeatureForm] when dealing with [UtilityAssociation]s.
+ *
+ * @since 300.0.0
+ */
+public sealed class FeatureFormNavigationRoute {
+
+    /**
+     * Indicates the route for the main [FeatureForm] screen. The associated [FeatureForm] can be
+     * obtained from the [FeatureFormState.activeFeatureForm] property.
+     *
+     * @since 300.0.0
+     */
+    public data object FeatureForm : FeatureFormNavigationRoute()
+
+    /**
+     * Indicates the route for [UtilityAssociationsFilterResult] screen.
+     *
+     * @param element The [UtilityAssociationsFormElement] associated with the filter result.
+     * @param utilityAssociationsFilterResult The selected [UtilityAssociationsFilterResult].
+     *
+     * @since 300.0.0
+     */
+    public data class FilterResult(
```

Comment:
I think we should prefix these with UtilityAssociation since all these destinations are UtilityAssociation related
```suggestion
    public data class UtilityAssociationFilterResult(
```

Reply:
The issue I see is we're repeating type information which makes it hard to read. This would make it read `UtilityAssociationFilterResult.utilityAssociationsFilterResult` which feels too verbose imo. I think we should keep the name short and succinct since these are just routes and not the actual types.

How about `AssociationsFilter`, `AssociationGroup` for these?